### PR TITLE
E_ALL implied E_NOTICE, hence remove the latter

### PR DESCRIPTION
**Pull request by MichaIng**
### Description:

Matomo enforces the error level, setting `E_ALL` and `E_NOTICE`. Since the first implies the second, the second is redundant. For reference: For reference: https://www.php.net/manual/en/errorfunc.configuration.php#ini.error-reporting

I can imagine that it was wanted to NOT show `E_NOTICE` logs, but for this the syntax is wrong and should then be:
```php
error_reporting(E_ALL & ~E_NOTICE);
```
For reference: https://www.php.net/manual/en/function.error-reporting.php#refsect1-function.error-reporting-examples

Also, I do not understand the real reason for this. It took me an hour to find out why I see deprecation warnings (testing PHP 8.2) even that I disabled them in `php.ini`. Is there any specific reason why Matomo enforces full error logging?

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by sgiehl** [comment]
Maybe the reason for setting the error level here is, that our own loggers are able to log any kind of error depending on the log configuration. But seems that "broken" error level has already been added a long time ago.
@tsteur Shall we simply remove the incorrect level, or do you think we should adjust anything there?

**Comment by MichaIng** [comment]
> Maybe the reason for setting the error level here is, that our own loggers are able to log any kind of error depending on the log configuration.

Ah, so a conflict between:
1. "When I change the log level in Matomo, I expect it to be effective."
2. "When I change the log level in `php.ini,` I expect it to be effective."

A bit a problem is that the different PHP warnings behind `E_*` is merged into one Matomo log level `WARN`. But differentiating this in Matomo makes it probably too complicated. I personally would like to have the PHP setting not overwritten, but instead a note in `config.ini.php` that PHP logs are shown in Matomo only when emitted by PHP based on `php.ini` `error_reporting` setting. On shared hostings, a `.user.ini` could be used to apply `E_ALL`, if needed.

A related question: Since I check PHP errors via `journalctl`, are there actually other logs done by Matomo or is it basically a duplication of PHP server logs?

**Comment by sgiehl** [comment]
@MichaIng In general I would agree. But many people using Matomo on a shared hosting might not be able to apply any changes to `php.ini`.
Matomo by default should write its logs to `matomo/tmp/logs/matomo.log`

**Comment by MichaIng** [comment]
> But many people using Matomo on a shared hosting might not be able to apply any changes to `php.ini`.

`.user.ini` should work. But I see the this would make things more complicated for shared hosting users, and my need to mute specific PHP warnings is probably an edge case as well.

> Matomo by default should write its logs to `matomo/tmp/logs/matomo.log`

Let me rephrase: Are all Matomo logs visible in PHP STDOUT/syslog (if enabled), or does Matomo do additional own logs, bypassing the PHP logging engine?

**Comment by tsteur** [comment]
It's actually there since the first commit :) https://github.com/matomo-org/matomo/blob/b23ec243e5d16852836fb7086a9150417acb2183/index.php#L5 

It should be fine to merge the PR assuming notices are still treated the same 👍 

I do vaguely remember that back then this was maybe needed to specifically add `E_NOTICE` but not sure.

**Comment by sgiehl** [comment]
@MichaIng All kind of errors triggered by PHP are handled with a custom error handler, so they will also end up in the Matomo log file. But Matomo might do some additional logging, that might not end up in php error logs.

@tsteur I'll merge this then. Maybe that was needed back in PHP 5 times or so. But with PHP 7.2+ there should be no need anymore.

**Comment by MichaIng** [comment]
> All kind of errors triggered by PHP are handled with a custom error handler, so they will also end up in the Matomo log file. But Matomo might do some additional logging, that might not end up in php error logs.

Okay, thanks for clarification. Would be actually great to have an option for not handling PHP errors, for cases where those are logged to journal/syslog/file by the PHP server already, and where admin has access to view those. Or the other way round, if possible an option to pass all Matomo logs through the PHP handler, so Matomo's own logs could be disabled and everything viewed via PHP/system logs instead. But that's another topic.
